lbrynet/lbrynet_daemon/LBRYDaemon.py

```diff
@@ -1897,43 +1902,38 @@ def _clean(n):
             t = []
             for i in n:
                 if i[0]:
-                    if i[1][0][0] and i[1][1][0] and i[1][2][0]:
-                        i[1][0][1]['value'] = str(i[1][0][1]['value'])
-                        t.append([i[1][0][1], i[1][1][1], i[1][2][1]])
+                    tr = {}
+                    tr.update(i[1][0])
+                    thumb = tr.get('thumbnail', None)
+                    if thumb is None:
+                        tr['thumbnail'] = "img/Free-speech-flag.svg"
+                    tr['cost_est'] = i[1][1]
+                    t.append(tr)
             return t
 
-        def resolve_claims(claims):
-            ds = []
-            for claim in claims:
-                d1 = defer.succeed(claim)
-                d2 = self._resolve_name(claim['name'])
-                d3 = self._get_est_cost(claim['name'])
-                dl = defer.DeferredList([d1, d2, d3], consumeErrors=True)
-                ds.append(dl)
-            return defer.DeferredList(ds)
-
-        def _disp(results):
-            log.info('Found ' + str(len(results)) + ' search results')
-            consolidated_results = []
-            for r in results:
-                t = {}
-                t.update(r[0])
-                if not 'thumbnail' in r[1].keys():
-                    r[1]['thumbnail'] = "img/Free-speech-flag.svg"
-                t.update(r[1])
-                t['cost_est'] = r[2]
-                consolidated_results.append(t)
-                # log.info(str(t))
-
-            return consolidated_results
+        def get_est_costs(results):
+            def _get_costs(search_result):
+                log.info("**" + search_result['name'])
+                d = self._get_est_cost(search_result['name'])
+                d.addCallback(lambda p: _save_cost(search_result, p))
+                return d
+
+            def _save_cost(value, cost):
+                log.info("Save cost")
+                log.info(value)
+                log.info(cost)
+                return [value, cost]
+
+            log.info("Estimating costs")
+            dl = defer.DeferredList([_get_costs(r) for r in results], consumeErrors=True)
+            return dl
 
         log.info('Search nametrie: ' + search)
 
         d = self._search(search)
         d.addCallback(lambda claims: claims[:self.max_search_results])
-        d.addCallback(resolve_claims)
+        d.addCallback(get_est_costs)
```

Comment:
Using nested functions in the callbacks is a tolerable pattern when they're tiny.  If a nested function has other nested functions, its becomes difficult to read and impossible to re-use.  In the future, consider making separate classes or un-nested functions.  Those can be tested and re-used.

Also, the scoping rules on nested functions become difficult to follow which leads to bugs.